src/app/modules/shared_models/emoji_reactions_model.nim

```diff
@@ -0,0 +1,84 @@
+import NimQml, Tables
+import emoji_reactions_item
+
+type
+    ModelRole {.pure.} = enum
+        EmojiId = UserRole + 1
+        Filename
+        DidIReactWithThisEmoji
+
+QtObject:
+    type Model* = ref object of QAbstractListModel
+        items: seq[EmojiReactionItem]
+
+    proc delete(self: Model) =
+        self.items = @[]
+        self.QAbstractListModel.delete
+
+    # TODO : To make this code scale, we can consider a loop similar to
+    # below code, and rename emoji to just be emojiReactions/emoji_[1 ... n]
+    #
+    # ```nim 
+    #   for i in 1..itemCount:
+    #       items.add(initItem(i, "emojiReactions/emoji_$(i)", false))
+    # ```
+    proc setup(self: Model) =
```

Comment:
It's a matter of taste, but I think it's better to fill the model from the outside.